deploy/charts/openebs-monitoring/rules/volumes/volume-node-rules.yaml

```diff
@@ -0,0 +1,47 @@
+groups:
+  - name: volume-node
+    rules:
+      # Alert for pv's fileystem getting into read-only state while re-mounting on a particular node.
```

Comment:
Should it be just mounting? Why should it be re-mounting?

Reply:
In the system logs it looks for the pattern: "Remounting filesystem read-only" for this condition. Also this document illustrates this condition. The volume was created and then the device was removed when the writes were going on. The pod using that was still running and all the writes thereon were rejected as the filesystem become read-only then.

Reply:
The document just talks about "Failed to mount..." Am I missing something?
cc @shubham14bajpai

Reply:
@avishnu @Ab-hishek is talking about the kmsg logs in the paste bin ref https://pastebin.com/YEMpfWkM. Here the error msg is `2,1533,18423953113,-;EXT4-fs (dm-0): Remounting filesystem read-only`.  The volume was already mounted on the application pod but the disk got detached and attached due to some reason, in this case the volume will remain mounted but I/O will fail due to the above error.

Reply:
But file system can go into read-only due to many reasons so `# Alert for pv's fileystem getting into read-only state ` should be enough as `problem_counters` don't have the error msg and just the reason.

Reply:
Ok.
What is the counter here? Is it the log message "Remounting filesystem read-only" ?

Reply:
Okay. Thanks Shubham for the clarification.

Reply:
@avishnu Yes
So NPD will have a config to look for patterns like this https://github.com/kubernetes/node-problem-detector/blob/70f79831ded13e9d39d50c61f8afeca34a68be43/config/kernel-monitor.json#L82. If the kmsg logs encounter that pattern it will increment the counter of that type by 1